src/compiler/checker.ts

```diff
@@ -14788,7 +14788,7 @@ namespace ts {
                     }
                 }
                 else {
-                    childrenTypes.push(checkExpression(child, checkMode));
+                    childrenTypes.push(checkExpressionForMutableLocation(child, checkMode));

```

Comment:
Why mutable location?

Reply:
jsxchildren are like a jsxattribute are like object members and should be checked the same way.